stdlib/public/stubs/Stubs.cpp

```diff
@@ -172,13 +172,13 @@ static uint64_t swift_floatingPointToString(char *Buffer, size_t BufferLength,
 extern "C" uint64_t swift_float32ToString(char *Buffer, size_t BufferLength,
                                           float Value) {
   return swift_floatingPointToString<float>(Buffer, BufferLength, Value,
-                                            "%0.*g");
+                                            "%0.9g");
```

Comment:
I'm not a C++ guru, so I have no opinion about the latter part but changing the format string is needed (too). %f defaults to 6 decimal places. The format string I suggested does not take two arguments.